File: src/services/FacturaService.js
// src/services/FacturaService.js
import axios from 'axios';

const FACTURA_API_URL = 'http://localhost:8080/api/facturas';

// Obtener todas las facturas
export const obtenerFacturas = () => {
  return axios.get(FACTURA_API_URL);
};

// Crear una factura nueva
export const crearFactura = (factura) => {
  return axios.post(FACTURA_API_URL, factura);
};

// Obtener una factura por ID
export const obtenerFacturaPorId = (id) => {
  return axios.get(`${FACTURA_API_URL}/${id}`);
};

// Actualizar factura
export const actualizarFactura = (id, factura) => {
  return axios.put(`${FACTURA_API_URL}/${id}`, factura);
};

// Eliminar factura
export const eliminarFactura = (id) => {
  return axios.delete(`${FACTURA_API_URL}/${id}`);
};

File: src/main/java/com/kajumboutique/kajumboutique/service/FacturaService.java
package com.kajumboutique.kajumboutique.service;

import java.util.List;
import java.util.Optional;

import org.springframework.stereotype.Service;

import com.kajumboutique.kajumboutique.model.Factura;
import com.kajumboutique.kajumboutique.repository.FacturaRepository;

@Service
public class FacturaService {

    private final FacturaRepository facturaRepository;

    public FacturaService(FacturaRepository facturaRepository) {
        this.facturaRepository = facturaRepository;
    }

    // Obtener todas las facturas
    public List<Factura> obtenerTodasLasFacturas() {
        return facturaRepository.findAll();
    }

    // Buscar factura por ID
    public Optional<Factura> obtenerFacturaPorId(int id) {
        return facturaRepository.findById(id);
    }

    // Guardar nueva factura
    public Factura guardarFactura(Factura factura) {
        return facturaRepository.save(factura);
    }

    // Actualizar factura existente
    public Factura actualizarFactura(Factura factura) {
        if (facturaRepository.existsById(factura.getId_factura())) {
            return facturaRepository.save(factura);
        }
        throw new RuntimeException("Factura no encontrada con id " + factura.getId_factura());
    }

    // Eliminar factura por ID
    public void eliminarFactura(int id) {
        facturaRepository.deleteById(id);
    }
}
